### DISCUSSION TITLE
The equivalent of StreamSelectDriver::getTimeout in EvDriver

### DISCUSSION THREAD

**Discussion by wtsergo** [revoltphp/event-loop#97] [Q&A] [answered]
Hello,

Is it possible to get timeout value in EvDriver that would be the equivalent of StreamSelectDriver::getTimeout?
I'm experimenting with adding mysqli async support in event loop.

Thanks,
Sergey Boyko

**Comment by trowski** [comment, marked as answer]
Not in a simple way, no.

There's not a good solution to poll mysql connections within the event loop because ext-mysqli doesn't expose a socket which can be polled like, for example, ext-pgsql.

The best solution is likely a short repeating timer which calls `mysqli::poll()` with either a very short or zero timeout.

Instead though I'd recommend you check out [`amphp/mysql`](https://github.com/amphp/mysql), which has IMO a better interface than ext-mysqli and has full async support.

**Reply by wtsergo** [reply]
Unfortunately amphp/mysql is extremely slow when doing mass insert/update operations.
PDO with amphp/parallel worker pool or customized event loop with mysqli much faster.
Eventually I did 3 level pool: 1st dozen of parallel PDO, 2nd hundred mysqli and 3rd amphp/mysql
The only problem with mysqli is that it abuse event loop idle state when using ev or uv driver because I can't query Driver::getTimeout
